### PR TITLE
Release 5.0.1

### PR DESCRIPTION
**Pull request by JEuler**

### CODE REVIEW

File: chopper_generator/pubspec.yaml [hunk 1]
```diff
@@ -8,7 +8,7 @@ environment:
   sdk: ">=2.17.0 <3.0.0"
 
 dependencies:
-  analyzer: ^4.1.0
+  analyzer: ">=4.1.0 <4.3.0"
```
Comment (techouse) [inline chopper_generator/pubspec.yaml:11]: This will have to be pumped to `^4.4.0`

File: chopper_generator/lib/src/generator.dart [hunk 2]
```diff
@@ -339,13 +361,9 @@ class ChopperGenerator extends GeneratorForAnnotation<chopper.ChopperApi> {
...
   }
 
-  /// TODO: upgrade analyzer to ^4.4.0 to replace enclosingElement with enclosingElement3
-  /// https://github.com/dart-lang/sdk/blob/main/pkg/analyzer/CHANGELOG.md#440
   String _factoryForFunction(FunctionTypedElement function) =>
-      // ignore: deprecated_member_use
-      function.enclosingElement is ClassElement
-          // ignore: deprecated_member_use
-          ? '${function.enclosingElement!.name}.${function.name}'
+      function.enclosingElement3 is ClassElement
```
Comment (techouse) [inline chopper_generator/lib/src/generator.dart:365]: Because of this analyzer needs to be `^4.4.0`

File: chopper_generator/pubspec.yaml (hunk 1 above)
Comment (JEuler) [reply chopper_generator/pubspec.yaml:11]: Whoops :)

**Review by techouse — APPROVED** [review]

**Comment by techouse** [comment]
Looks like you forgot to bump the versions and the deployment failed.

**Comment by JEuler** [comment]
Fucking (sorry) rebase worked not as I think it should.

**Comment by JEuler** [comment]
>Maybe a dart pub publish --dry-run test needs to go into the unit test suite for deployment / master merges?
Good Idea, btw! :)

**Pull request merged**